activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/SecurityPoliciesApplicationService.java

```diff
@@ -76,19 +69,27 @@ private boolean noSecurityPoliciesOrNoUser() {
     }
 
     public ProcessInstanceQuery restrictProcessInstQuery(ProcessInstanceQuery query, SecurityPolicy securityPolicy){
+        return restrictQuery(query, processInstanceRestrictionApplier, securityPolicy);
+    }
+
+    private  <T extends Query<?,?>> T restrictQuery(T query, SecurityPoliciesRestrictionApplier<T> restrictionApplier, SecurityPolicy securityPolicy){
         if (noSecurityPoliciesOrNoUser()){
             return query;
         }
 
         Set<String> keys = definitionKeysAllowedForRBPolicy(securityPolicy);
 
         if(keys != null && !keys.isEmpty()){
-            return query.processDefinitionKeys(keys);
+
+            if(keys.contains(securityPoliciesService.getWildcard())){
+                return query;
+            }
+
+            return restrictionApplier.restrictToKeys(query, keys);
         }
 
         if((keys != null || !keys.isEmpty()) && securityPoliciesService.policiesDefined()){
```

Comment:
@ryandawsonuk I'm little bit confused about the first part of this if.. it's going produce a null pointer exception if keys is null.